src/scraper/dmhy.ts

```diff
@@ -60,128 +62,149 @@ export class DmhyScraper implements Scraper {
-            return anchor.getAttribute('href');
-        }, btResourceElement));
-        item.magnet_uri = await page.evaluate(el => {
-            const anchor = el.querySelector('#a_magnet');
-            return anchor.getAttribute('href');
-        }, btResourceElement);
+            let resourceInfoElement = await mainArea.$('.info.resource-info.right > ul');
+            item.timestamp = toUTCDate(await page.evaluate(el => {
+                const spanEl = el.querySelector('li:nth-child(2) > span');
+                let timeStr = spanEl.textContent;
+                if (timeStr) {
+                    return timeStr.trim();
+                }
+                return Date.now();
+            }, resourceInfoElement), 8);
+            // console.log(item.timestamp);
+            let btResourceElement = await mainArea.$('#tabs-1');
+            item.torrent_url = resolve('https://', await page.evaluate(el => {
+                const anchor = el.querySelector('p:nth-child(1) > a');
+                return anchor.getAttribute('href');
+            }, btResourceElement));
+            item.magnet_uri = await page.evaluate(el => {
+                const anchor = el.querySelector('#a_magnet');
+                return anchor.getAttribute('href');
+            }, btResourceElement);
 
-        item.files = [];
-        let fileElHandlerList = await btResourceElement.$$('.file_list > ul > li');
-        for (let fileElHandler of fileElHandlerList) {
-            let mediaFile = new MediaFile();
-            mediaFile.size = await page.evaluate(
-                el => el.querySelector('.bt_file_size').textContent.trim(), fileElHandler);
-            mediaFile.path = await page.evaluate(el => el.textContent.trim(), fileElHandler);
-            let match = mediaFile.path.match(/(.+?)\t/);
-            if (match) {
-                mediaFile.path = match[1];
+            item.files = [];
+            let fileElHandlerList = await btResourceElement.$$('.file_list > ul > li');
+            for (let fileElHandler of fileElHandlerList) {
+                let mediaFile = new MediaFile();
+                mediaFile.size = await page.evaluate(
+                    el => el.querySelector('.bt_file_size').textContent.trim(), fileElHandler);
+                mediaFile.path = await page.evaluate(el => el.textContent.trim(), fileElHandler);
+                let match = mediaFile.path.match(/(.+?)\t/);
+                if (match) {
+                    mediaFile.path = match[1];
+                }
+                mediaFile.ext = extname(mediaFile.path);
+                mediaFile.name = basename(mediaFile.path, mediaFile.ext);
+                item.files.push(mediaFile);
             }
-            mediaFile.ext = extname(mediaFile.path);
-            mediaFile.name = basename(mediaFile.path, mediaFile.ext);
-            item.files.push(mediaFile);
+            return item;
+        } catch (e) {
+            console.error(e);
+            return null;
```

Comment:
why add return?

Reply:
The function should return something when a exception is raised. explicity return has better readiblity.